### PR TITLE
Scan Oclint integration

### PR DESCRIPTION
**Pull request by andybest**
This pull request addresses issues #4327 and #1934, and allows integration between the Scan and OCLint actions. It also fixes an issue that I found with the current OCLint action.

OCLint requires (at least the current version does) the path of the directory containing the compilation database. The current OCLint action in Master instead provides the path to the actual file, which doesn't work. I have rectified this.

I also added an extra configuration parameter (`use_clang_report_name`) to Scan to output the compilation database JSON with the naming convention that OCLint expects [http://clang.llvm.org/docs/JSONCompilationDatabase.html](http://clang.llvm.org/docs/JSONCompilationDatabase.html). I decided to add this as a config parameter rather than the default to avoid breaking any existing workflows that may use the current naming of this file.

Basically, this means that a Scan + OCLint workflow could look like this:

```
fastlane_version "1.89.0"

default_platform :ios

platform :ios do
  desc "Runs all the tests"
  lane :test do
    scan(
        clean: true,
        scheme: "MyApp",
        output_types: "json-compilation-database",
        use_clang_report_name: true
    )

    oclint(
        compile_commands: Dir.pwd + "/test_output/compile_commands.json",
        report_type: 'pmd'
    )
  end
end
```

### CODE REVIEW

**Comment by andybest** [comment]
That's strange, hound told me to use slash syntax around regular expressions in another pull request, now it's telling me to use %r?!?

**Comment by andybest** [comment]
The CircleCI failure also doesn't seem to be anything to do with the tests?

**Comment by mfurtak** [comment]
Hi @andybest - thanks for the contribution! 👍  I'll take a look.

File: fastlane/lib/fastlane/actions/oclint.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def self.run(params)
         oclint_args << "-enable-clang-static-analyzer" if params[:enable_clang_static_analyzer]
         oclint_args << "-enable-global-analysis" if params[:enable_global_analysis]
         oclint_args << "-allow-duplicated-violations" if params[:allow_duplicated_violations]
-        oclint_args << "-p #{params[:compile_commands]}" if params[:compile_commands]
+        oclint_args << "-p #{File.dirname(params[:compile_commands])}" if params[:compile_commands]
```
Comment (mfurtak) [inline fastlane/lib/fastlane/actions/oclint.rb]: This doesn't seem backwards compatible for people's existing configurations. What's the reasoning behind this change?

Comment (andybest) [inline fastlane/lib/fastlane/actions/oclint.rb]: OCLint requires the _directory_ of the compile commands file, not the file itself. Without this change, it does not work at all. I'm not sure if this is a recent change in OCLint.

Comment (TiagoVeloso) [inline fastlane/lib/fastlane/actions/oclint.rb]: I've took this from the docs:

> -p <build-path>
> Choose the build directory in which a file named compile_commands.json exists. When no build path is specified, > a search for compile_commands.json will be attempted through all parent paths of the first input file.

http://docs.oclint.org/en/stable/manual/oclint.html#compilation-database

However, I don't know how this translates to this fastlane action. I've been waiting for `scan` support to start using OCLint, so I can't say whether the action is working or not at it is on `master`

Comment (andybest) [inline fastlane/lib/fastlane/actions/oclint.rb]: The fastlane action takes the path of the actual file, so it can parse it itself to get the filenames out. It was originally passing this to OCLint with the -P option, which requires the _directory_ that the file resides in, so it wasn't working on master.

Comment (mfurtak) [inline fastlane/lib/fastlane/actions/oclint.rb]: Got it. Thanks for the background! 👍

**Comment by mfurtak** [comment]
Hi @andybest - I'm working on bringing this in through #4992. Can you let me know how that looks to you?

**Comment by andybest** [comment]
@mfurtak Commented directly on your PR- looks good!

**Comment by mfurtak** [comment]
Excellent! This is merged as part of https://github.com/fastlane/fastlane/pull/4992 and will go out with the next release :rocket: Thanks!

**Comment by mfurtak** [comment]
This is released as part of [`scan` 0.8.0](https://github.com/fastlane/fastlane/releases/tag/scan%2F0.8.0) 🎉
